OpenRA.Game/Map/MapGridData.cs

```diff
@@ -0,0 +1,61 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using System.IO;
+
+namespace OpenRA
+{
+	public enum TileShape { None, Rectangle, Diamond }
+
+	public class MapGridData
```

Comment:
Is there any specific reason for having separate `MapGridData` and `MapGrid` types?
If you combine the two you can use `FieldLoader` instead of manually loading the fields from yaml.